src/api/test/activexml_matcher.rb

```diff
@@ -175,7 +175,10 @@ def self.match(node, conditions)
     if conditions[:ancestor]
       return false unless catch :found do
         p = node
-        throw :found, true if match(p, conditions[:ancestor]) while p = p.parent
+        while p
+          p = p.parent
```

Comment:
I think that yours is better than mine. I'll change it. 
In my implementation *p* can be *nil* when calling *match*, and can be wrong.